### PR TITLE
DRIVERS-2395: Add test that read is not retried in a transaction

### PR DESCRIPTION
**Pull request by jyemin**
Please complete the following before merging:
- [N/A] Bump spec version and last modified date.
- [N/A] Update changelog.
- [Done] Make sure there are generated JSON files from the YAML test files.
- [Done] Test changes in at least one language driver.
- [Done] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

Though the requirement is in the retryable reads spec, the test seemed more appropriate in the transactions spec.  Happy to move if there is strong disagreement.

DRIVERS-2395

### CODE REVIEW

**Comment by jyemin** [comment]
Huh, this is failing on 4.0 variants with 

```
{"ok": 0.0, "errmsg": "failCommand not found", "code": 266, "codeName": "FailPointSetFailed"
```

I'm not sure why, it's passing for me locally on a 4.0.18 replica set.

**Comment by ShaneHarvey** [comment]
Should we add a test for reads too or does that already exist?

**Comment by jyemin** [comment]
I removed [this check](https://github.com/mongodb/mongo-java-driver/blob/5c903986d5f5330dd390f1c354e7c746a2337142/driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java#L413-L415) and a bunch of tests fail, including [this one](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/legacy/retryable-writes.json#L254).  So I think we're covered for writes.

**Comment by jyemin** [comment]
Added errorLabelsContain assertion to match what's in the corresponding test for writes.

**Review by ShaneHarvey — APPROVED** [review]

File: source/transactions/tests/unified/do-not-retry-read-in-transaction.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
+{
+  "description": "do not retry read in a transaction",
+  "schemaVersion": "1.4",
+  "runOnRequirements": [
+    {
+      "minServerVersion": "4.0.0",
+      "topologies": [
+        "single",
```
Comment (jyemin) [inline source/transactions/tests/unified/do-not-retry-read-in-transaction.json:8]: I errantly added this topology, and didn't wait for tests to complete on Evergreen.  Will remove in subsequent commit.

Comment (jmikola) [reply source/transactions/tests/unified/do-not-retry-read-in-transaction.json:8]: https://github.com/mongodb/specifications/commit/bafb024be8202b1676a3c6632e8a3eb9cf36209b

**Pull request merged**